src/main/java/org/spongepowered/api/text/dialogue/DialogueArchetype.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataQuery;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Collection;
+
+/**
+ * Represents a structure for a back-and-forth communication between one or
+ * more {@link Speaker}s and one or more plugiins.
+ */
+public interface DialogueArchetype {
+
+    /**
+     * Creates a new {@link Builder} to build a new {@link DialogueArchetype}.
+     *
+     * @return The newly created builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Speaker... speakers);
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Iterable<? extends Speaker> speakers);
+
+    /**
+     * Gets the first question that is shown to the {@link Speaker} when the
+     * {@link Dialogue} is instantiated.
+     *
+     * @return The initial question
+     */
+    Question getInitialQuestion();
+
+    /**
+     * Gets the initial data that the generated {@link Dialogue} will be
+     * populated with. This is a copy of the data.
+     *
+     * @return The data
+     */
+    DataContainer getInitialData();
+
+    /**
+     * Gets whether or not output is suppressed by default.
+     *
+     * @return Whether output is suppressed
+     * @see Builder#suppressesOutput(boolean)
+     */
+    boolean suppressesOutput();
+
+    /**
+     * Gets whether or not messages are suppressed by default.
```

Comment:
You should add what actually happens if messages are suppressed. Aka `the speakers will only recieve chat from X. Any other message will be deleted/delayed...`

The same goes for the other boolean methods.

Please add an option to just delay chat. So that it will be send after the dialog is done.